src/devtools/client/debugger/src/components/Editor/ColumnBreakpoints.tsx

```diff
@@ -92,29 +100,19 @@ function PointsForRow({
   editor,
   points,
 }: {
-  breakpoints: SameLineSourceLocations[];
+  breakpoints: Map<number, SameLineSourceLocations>;
   editor: SourceEditor;
   points: Point[];
 }) {
   const line = points[0]!.location.line;
 
-  // TODO BAC-2329
-  // The backend sometimes returns duplicate columns per line;
-  // In order to prevent the frontend from showing something weird, let's dedupe them here.
-  const breakpointsForLine = useMemo(() => {
-    const match = breakpoints.find(breakpoint => breakpoint.line === line);
```

Comment:
This was probably a bit silly before. If the breakpoints returned by the backend _are_ pre-sorted (as the comment elsewhere says) we should have been using a binary search here rather than a linear one.